ovirt/resource_ovirt_vm.go

```diff
@@ -1609,3 +1631,42 @@ func getTemplateDiskAttachments(templateID string, meta interface{}) ([]*ovirtsd
 	}
 	return nil, nil
 }
+
+func getAffinityGroups(conn *ovirtsdk4.Connection, cID string, agNames []interface{}) (ag []*ovirtsdk4.AffinityGroup, err error) {
+	var ags []*ovirtsdk4.AffinityGroup
+	res, err := conn.SystemService().ClustersService().
+		ClusterService(cID).AffinityGroupsService().
+		List().Send()
+	if err != nil {
+		return nil, err
+	}
+	agNamesMap := make(map[string]*ovirtsdk4.AffinityGroup)
+	for _, af := range res.MustGroups().Slice() {
+		agNamesMap[af.MustName()] = af
+	}
+	for _, agName := range agNames {
+		if _, ok := agNamesMap[agName.(string)]; !ok {
+			return nil, fmt.Errorf("affinity group %v was not found on cluster %v", agName, cID)
```

Comment:
Please build the error based on the entire loop processing rather than exiting on the first error

Reply:
Fixed this is an example of the error

Error: affinity groups [test5 test6 test9] was not found on cluster 07716001-e669-4ded-b2e7-099543a0446e

Reply:
lgtm